### PR TITLE
[202012][dhcp-relay] Add support for dhcp_relay config cli

### PR DESCRIPTION
**Pull request by yaqiangz**
#### Why I did it
Currently the config cli of dhcpv4 is may cause confusion and config of dhcpv6 is missing.

#### How I did it
Add dhcp_relay config cli and test cases.
```
config dhcp_relay ipv4 helper (add | del) <vlan_id> <helper_ip_list>
config dhcp_relay ipv6 destination (add | del) <vlan_id> <destination_ip_list>
```

#### How to verify it
1. Verify by unitest, all passed
```
2023-01-31T17:13:39.5310534Z ----------- coverage: platform linux, python 3.7.3-final-0 -----------
2023-01-31T17:13:39.5311171Z Name                                          Stmts   Miss  Cover
2023-01-31T17:13:39.5311934Z -----------------------------------------------------------------
2023-01-31T17:13:39.5318510Z config/dhcp_relay.py                            101     11    89%
2023-01-31T17:13:39.5375220Z utilities_common/dhcp_relay_util.py               7      0   100%
2023-01-31T17:13:39.5378676Z -----------------------------------------------------------------
2023-01-31T17:13:39.5379423Z Coverage HTML written to dir htmlcov
2023-01-31T17:13:39.5379772Z Coverage XML written to file coverage.xml
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml
```
2. Build sonic-utilities and install in DUT and run cli, test cli
```
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv6 destination add 1000 fc02:2000::6    
Added DHCP relay address [fc02:2000::6] to Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv6 destination add 1000 fc02:2000::7 fc02:2000::8
Added DHCP relay address [fc02:2000::7,fc02:2000::8] to Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ show dhcprelay_helper ipv6                                                
--------  ------------
Vlan1000  fc02:2000::1
          fc02:2000::2
          fc02:2000::3
          fc02:2000::4
          fc02:2000::6
          fc02:2000::7
          fc02:2000::8
--------  ------------
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv6 destination del 1000 fc02:2000::6   
Removed DHCP relay address [fc02:2000::6] from Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv6 destination del 1000 fc02:2000::7 fc02:2000::8   
Removed DHCP relay address [fc02:2000::7,fc02:2000::8] from Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ show dhcprelay_helper ipv6                                                
--------  ------------
Vlan1000  fc02:2000::1
          fc02:2000::2
          fc02:2000::3
          fc02:2000::4
--------  ------------
```

```
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv4 helper add 1000 192.0.0.49
Added DHCP relay address [192.0.0.49] to Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv4 helper add 1000 192.0.0.50 192.0.0.51
Added DHCP relay address [192.0.0.50,192.0.0.51] to Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv4 helper del 1000 192.0.0.49   
Removed DHCP relay address [192.0.0.49] from Vlan1000
Restarting DHCP relay service...
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv4 helper del 1000 192.0.0.49
Usage: config dhcp_relay ipv4 helper del [OPTIONS] <vid> DHCP_RELAY_HELPERS...
Try "config dhcp_relay ipv4 helper del -h" for help. Error: 192.0.0.49 is not a DHCP relay for Vlan1000
admin@str-e1031-acs-3:~$ sudo config dhcp_relay ipv4 helper del 1000 192.0.0.50 192.0.0.51   
Removed DHCP relay address [192.0.0.50,192.0.0.51] from Vlan1000
Restarting DHCP relay service...
```

### CODE REVIEW

**Comment by yaqiangz** [comment]
@kellyyeh @jcaiMR @Blueve Could you please help to review this PR?

File: config/dhcp_relay.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        dhcp_servers.append(dhcp_relay_ip)
+        added_ips.append(dhcp_relay_ip)
+
+    table[dhcp_servers_str] = dhcp_servers
+
+    db.cfgdb.set_entry(table_name, vlan_name, table)
+    click.echo("Added DHCP relay address [{}] to {}".format(",".join(dhcp_relay_ips), vlan_name))
+    try:
+        dhcp_relay_util.restart_dhcp_relay_service()
```
Comment (Blueve) [inline config/dhcp_relay.py:64]: extract to `def handle_restart_dhcp_relay_service()`?

File: tests/config_dhcp_relay_test.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    return request.param
+
+
+class TestConfigDhcpRelay(object):
+    @classmethod
+    def setup_class(cls):
+        os.environ['UTILITIES_UNIT_TESTING'] = "1"
+        print("SETUP")
+
+    def test_config_dhcp_relay_add_del_with_nonexist_vlanid(self, ip_version, op):
```
Comment (Blueve) [inline tests/config_dhcp_relay_test.py:124]: Do we have code to handle situation like `remove_vlan_while_dhcp_config_exist`?

File: config/dhcp_relay.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        ctx.fail("{} doesn't exist".format(vlan_name))
+
+    dhcp_servers = table.get(dhcp_servers_str, [])
+
+    return dhcp_servers, table
+
+
+def add_dhcp_relay(vid, dhcp_relay_ips, db, ip_version):
+    table_name = DHCP_RELAY_TABLE if ip_version == 6 else VLAN_TABLE
+    dhcp_servers_str = DHCPV6_SERVERS if ip_version == 6 else DHCPV4_SERVERS
```
[the remark this answers is not in the document]
Comment (yaqiangz) [reply config/dhcp_relay.py:39]: Fixed

File: tests/config_dhcp_relay_test.py (hunk 2 above)
Comment (yaqiangz) [reply tests/config_dhcp_relay_test.py:124]: Fixed

File: config/dhcp_relay.py (hunk 1 above)
Comment (yaqiangz) [reply config/dhcp_relay.py]: Fixed

File: tests/config_dhcp_relay_test.py (hunk 2 above)
Comment (yaqiangz) [reply tests/config_dhcp_relay_test.py:124]: Fixed by https://github.com/sonic-net/sonic-utilities/pull/2660

**Review by kellyyeh — APPROVED** [review]

**Review by Blueve — APPROVED** [review]

**Pull request merged**